### DISCUSSION TITLE
Long term caching of requests

### DISCUSSION THREAD

**Discussion by sidharthv96** [encode/httpx#1693] [Q&A] [answered]
Hi,
I'm calling some very expensive API's whose responses don't change over time.
I was using https://github.com/reclosedev/requests-cache with requests, but needed async support, so switched to httpx.

My first thought was to use the event_hooks to check a cache and return if a value is found. But when I checked the implementation, it didn't seem possible.

Is there any other way I can implement this?

**Comment by lovelydinosaur** [comment, marked as answer]
Maybe try taking a look at @johtso's [httpx-caching](https://github.com/johtso/httpx-caching) package.

I've not tried it yet myself, so I don't know what state it's in. Also we oughta get it added to [the third party docs](https://www.python-httpx.org/third_party_packages/).

**Reply by sidharthv96** [reply]
Thanks @tomchristie !
Seems like the package is not compatible with v0.18 and unmaintained.
But a very good starting point for me to fork from.

**Reply by johtso** [reply]
@sidharthv96 I put most of the work into updating it for 0.18 before it was officially released, but never revisited it once the new version was cut: https://github.com/johtso/httpx-caching/commits/httpx-v0.18

Will try and take a look over the next couple of days.
